educator/src/Dscp/Educator/Web/Educator/Queries.hs

```diff
@@ -248,3 +253,21 @@ educatorPostGrade subH grade = do
             , trIdx = TxInMempool
             , trSubmission = packPk subH
             }
+
+storeCertificateOlolo :: CertificateMeta -> PDFBody -> m ()
+storeCertificateOlolo _ _ = error "Implement"
+
+educatorAddCertificate
+    :: MonadEducatorWeb ctx m
+    => CertificateFullInfo -> m ()
+educatorAddCertificate cert = do
+    mblk <- transact $ createPrivateBlockFromCert cert
+    -- This private block will be added to the public chain on itself shortly
+
+    eAddr <- ourAddress @EducatorNode
+    let blk = mblk ?: error "What if there were no grades provided?"
+    let sAddr = error "Where do we get it? Shouldn't it be in CertificateMeta?"
```

Comment:
As I noticed in the previous comment, we cannot meaningfully assign an address to a student if a certificate is created solely by Educator, so we should use some hardcoded address (or Educator's address, or something similar which makes it possible to distinguish between genuine Student's signatures and "fake" ones provided by Educator)

Reply:
Doing it, although I find it a bit weird, since if an educator knows a student name then he probably should know the student address as well.
And student address is the only viable identifier; would be funny to produce a certificate which applies to multiple people.

Reply:
@Martoon-00 well, there are _no_ real student's addresses in our use case; does ITMO know your Disciplina address? You don't even have one.

The certificate generator is for Educators willing to provide verifiable versions of diplomas for existing students. Those students do not have a Disciplina address, and Educators definitely do not currently want to ask their students to obtain a Disciplina address and sign those grades. They could generate the Disciplina keys for students themselves and send those keys to the students together with certificates, but this kinda misses the point of a _secret_ key - moreover, it's possible for some student to receive two different secret keys from two different educators this way.

So it's better to not require a distinguished secret key for every student at all for this usecase, but instead make it possible to easily distinguish between a FairCV created "properly" - via Student App and containing signed submissions from a student - and a FairCV created "improperly" - solely by Educator via certificate generator.

Reply:
Thanks, now it makes sense to me.